client_test.go

```
@@ -278,12 +278,18 @@ func TestNewClientClonesMutableAMQPConfigContainers(t *testing.T) {
 	client := NewClient(cfg)
 	t.Cleanup(func() { _ = client.Close() })
 
+	auth.Username = "changed"
+	auth.Password = ""
 	cfg.AMQP.SASL[0] = &amqp.PlainAuth{Username: "changed"}
 	cfg.AMQP.Properties["product"] = "changed"
 	cfg.AMQP.TLSClientConfig.ServerName = "changed.example"
 
-	if client.config.AMQP.SASL[0] != auth {
-		t.Fatal("client SASL slice changed with caller slice")
+	clientAuth := client.config.AMQP.SASL[0].(*amqp.PlainAuth)
```

golangci-lint (2.12.2)

[error] type assertion must be checked

(forcetypeassert)